includes/assets/js/NFDPluginsMarketplace.js

```diff
@@ -0,0 +1,186 @@
+    }
+
+    // Filter products to only show plugins
+    filterProducts(products) {
+        const plugins = products.filter(product => product.type == "plugin");
+
+        return plugins;
+    }
+
+    // Show the filtered products on the page
+    showProducts(products) {
+        
+        // Helper function to render the thumbnail for each product
+        const renderThumbnail = (product) => {
+            if (product.smallThumbnailUrl) {
+                return `<img src="${product.smallThumbnailUrl}" alt="${product.name} Thumbnail" width="128px" height="128px">`;
+            }
+
+            const fallback = 'https://hiive-space-cdn.nyc3.cdn.digitaloceanspaces.com/marketplace/premium-plugins/fallback.svg';
+            return `<img src="${fallback}" alt="${product.name} Thumbnail" width="128px" height="128px">`;
+        }
+
+        // Helper function to render the action buttons for each product
+        const renderActions = (product) => {
+            const supportsCTB = window.supportsCTB;
+            let actionButtons = [];
+
+            if (product.primaryUrl && product.price) {
+                const primaryAction = `<a target="_blank" 
+                href="${product.primaryUrl}" 
+                class="button button-primary nfd-ctb-btn"
+                ${supportsCTB && product.clickToBuyId ? 'data-action="load-nfd-ctb"' + 'data-ctb-id="' + product.clickToBuyId + '"' : ''}>Buy $${product.price}</a>`;
+
+                actionButtons.push(primaryAction);
+            }
+
+            if (product.secondaryUrl) {
+                const secondaryAction = `<a target="_blank" href="${product.secondaryUrl}">More Details</a>`;
+
+                actionButtons.push(secondaryAction);
+            }
+            
+            return actionButtons.join('');
+        }
+        
+        // Helper function to render a template for each product
+        const buildProduct = (product) => {
+            return `<div class="plugin-card">
+                <div class="nfd-plugin-card-content">
+                    <div class="nfd-plugin-card-thumbnail">
+                        ${renderThumbnail(product)}
+                    </div>
+                    <div class="nfd-plugin-card-title name column-name">
+                        <h3>${product.name}</h3>
+                    </div>
+                    <div class="nfd-plugin-card-actions action-links">
+                        ${renderActions(product)}
+                    </div>
+                    <div class="nfd-plugin-card-desc desc column-description">
+                        <p>${product.description}</p>
+                    </div>
+                </div>
+            </div>`;
+        }
+
+        const render = products.map(product => buildProduct(product)).join('');
+
+        const pluginsListTable = document.querySelector('.wp-list-table #the-list');
+        return pluginsListTable.innerHTML = render;
```

Comment:
I know this content comes from us, but seems best to sanitize this innerHTML. I wonder what sanitizing methods are available in wp js libraries already.

Reply:
@circlecube Thanks for reviewing. Just a small follow-up, I couldn't find anything in core or from an official library to tackle this case; closest I found is this: https://github.com/cure53/DOMPurify

Would this work for this scenario?